### PR TITLE
[Modules] logic apps default values#784

### PR DESCRIPTION
**Pull request by prasanjeets**
# Description
Accommodated security for below policies  #784 
\built-in-policies\policyDefinitions\Logic Apps\LogicApps_AuditDiagnosticLog_Audit.json
\built-in-policies\policyDefinitions\Logic Apps\LogicApps_ISEWithCustomerManagedKey_AuditDeny.json
\built-in-policies\policyDefinitions\Azure Government\Logic Apps\LogicApps_ISEWithCustomerManagedKey_AuditDeny.json

| Pipeline |
| - |
|[Logic: Workflows](https://github.com/prasanjeets/ResourceModules/actions/workflows/ms.logic.workflows.yml) |

# Type of Change

Please delete options that are not relevant.

- [x] Bugfix (non-breaking change which fixes an issue)

# Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: modules/Microsoft.Logic/workflows/readme.md [hunk 1]
```diff
@@ -7,16 +7,17 @@ This module deploys a Logic App resource.
...
+- [Cross-referenced modules](#Cross-referenced-modules)
 - [Deployment examples](#Deployment-examples)
 
 ## Resource types
 
-| Resource Type | API Version |
-| :-- | :-- |
-| `Microsoft.Authorization/locks` | [2017-04-01](https://docs.microsoft.com/en-us/azure/templates/Microsoft.Authorization/2017-04-01/locks) |
+| Resource Type                             | API Version                                                                                                                       |
+| :---------------------------------------- | :-------------------------------------------------------------------------------------------------------------------------------- |
```
Comment (MrMCake) [inline modules/Microsoft.Logic/workflows/readme.md:16]: It is very likely this autoformatting will fail the pipeline. Please undo all occurences  (`-` & spaces)

**Review by MrMCake — CHANGES_REQUESTED** [review]
Please resolve the merge conflict

File: modules/Microsoft.Logic/workflows/readme.md (hunk 1 above)
Comment (prasanjeets) [reply modules/Microsoft.Logic/workflows/readme.md]: @MrMCake 
Made the requested changes, let me know if its correct

**Comment by prasanjeets** [comment]
>
@MrMCake Please check, I have pulled from main.

File: modules/Microsoft.Logic/workflows/deploy.bicep [hunk 2]
```diff
@@ -25,8 +25,8 @@ param userAssignedIdentities object = {}
 @description('Optional. The integration account.')
 param integrationAccount object = {}
 
-@description('Optional. The integration service environment.')
-param integrationServiceEnvironment object = {}
+@description('Optional. The integration service environment Id.')
+param integrationServiceEnvironmentId string = ''
```
Comment (MrMCake) [inline modules/Microsoft.Logic/workflows/deploy.bicep:29]:
```suggestion
param integrationServiceEnvironmentResourceId string = ''
```

File: modules/Microsoft.Logic/workflows/deploy.bicep [hunk 3]
```diff
@@ -172,7 +172,10 @@ resource logicApp 'Microsoft.Logic/workflows@2019-05-01' = {
       workflowManagement: !empty(workflowManagementAccessControlConfiguration) ? workflowManagementAccessControlConfiguration : null
     }
     integrationAccount: !empty(integrationAccount) ? integrationAccount : null
-    integrationServiceEnvironment: !empty(integrationServiceEnvironment) ? integrationServiceEnvironment : null
+    integrationServiceEnvironment: !empty(integrationServiceEnvironmentId) ? {
```
Comment (MrMCake) [inline modules/Microsoft.Logic/workflows/deploy.bicep:175]:
```suggestion
    integrationServiceEnvironment: !empty(integrationServiceEnvironmentResourceId) ? {
```

File: modules/Microsoft.Logic/workflows/readme.md [hunk 4]
```diff
@@ -29,37 +29,36 @@ This module deploys a Logic App resource.
...
-| `enableDefaultTelemetry` | bool | `True` |  | Enable telemetry via the Customer Usage Attribution ID (GUID). |
-| `integrationAccount` | object | `{object}` |  | The integration account. |
-| `integrationServiceEnvironment` | object | `{object}` |  | The integration service environment. |
-| `location` | string | `[resourceGroup().location]` |  | Location for all resources. |
+| `diagnosticSettingsName` | string | `[format('{0}-diagnosticSettings', parameters('name'))]` | | The name of the diagnostic setting, if deployed. |
+| `diagnosticStorageAccountId` | string | `''` | | Resource ID of the diagnostic storage account. |
+| `diagnosticWorkspaceId` | string | `''` | | Resource ID of the diagnostic log analytics workspace. |
+| `enableDefaultTelemetry` | bool | `True` | | Enable telemetry via the Customer Usage Attribution ID (GUID). |
+| `integrationAccount` | object | `{object}` | | The integration account. |
+| `integrationServiceEnvironmentId` | string | `''` | | The integration service environment Id. |
```
Comment (MrMCake) [inline modules/Microsoft.Logic/workflows/readme.md:46]:
```suggestion
| `integrationServiceEnvironmentResourceId` | string | `''` | | The integration service environment Id. |
```

**Review by MrMCake — APPROVED** [review]

**Pull request merged**